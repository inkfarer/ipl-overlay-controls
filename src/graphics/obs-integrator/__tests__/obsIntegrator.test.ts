import ObsIntegrator from '../obsIntegrator.vue';
import { config, mount } from '@vue/test-utils';

describe('ObsIntegrator', () => {
    config.global.stubs = {
        FontAwesomeIcon: true
    };

    beforeEach(() => {
        window.obsstudio = undefined;
    });

    it('matches snapshot outside obs', () => {
        const wrapper = mount(ObsIntegrator);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot within obs', () => {
        // @ts-ignore
        window.obsstudio = { pluginVersion: '2.18.9' };

        const wrapper = mount(ObsIntegrator);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot if obs version is too low', () => {
        // @ts-ignore
        window.obsstudio = { pluginVersion: '2.10.0' };

        const wrapper = mount(ObsIntegrator);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('hides message after timeout when obs version is too low', async () => {
        jest.useFakeTimers();
        // @ts-ignore
        window.obsstudio = { pluginVersion: '2.10.0' };

        const wrapper = mount(ObsIntegrator);

        expect(wrapper.findComponent('[data-test="outdated-plugin-warning"]').exists()).toEqual(true);
        jest.advanceTimersByTime(60000);
        await wrapper.vm.$nextTick();
        expect(wrapper.findComponent('[data-test="outdated-plugin-warning"]').exists()).toEqual(false);
    });
});
