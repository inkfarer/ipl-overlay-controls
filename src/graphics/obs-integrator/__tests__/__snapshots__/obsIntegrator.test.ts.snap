// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObsIntegrator matches snapshot if obs version is too low 1`] = `
"<div class=\\"layout vertical center-horizontal\\">
  <div class=\\"ipl-message__wrapper layout horizontal center-vertical ipl-message__type-warning width-cap m-t-8\\" data-test=\\"outdated-plugin-warning\\" data-v-13b37b8d=\\"\\">
    <font-awesome-icon-stub icon=\\"exclamation-triangle\\" border=\\"false\\" fixedwidth=\\"false\\" listitem=\\"false\\" pulse=\\"false\\" swapopacity=\\"false\\" spin=\\"false\\" symbol=\\"false\\" inverse=\\"false\\" class=\\"icon\\" data-v-13b37b8d=\\"\\"></font-awesome-icon-stub>
    <div class=\\"message-content\\" data-v-13b37b8d=\\"\\">
      <!---->
      <div class=\\"title\\">Outdated OBS version!</div> The installed OBS browser source plugin is out of date.<br> Integrations may not work on this version of OBS Studio.<br> Please update your OBS Studio version.
    </div>
  </div>
</div>"
`;

exports[`ObsIntegrator matches snapshot outside obs 1`] = `
"<div class=\\"layout vertical center-horizontal\\">
  <div class=\\"ipl-space color-dark width-cap\\" data-v-11398c0e=\\"\\">
    <div class=\\"title\\">IPL-OC obs-integrator</div> Integrates with OBS Studio to automate common tasks during broadcast.
  </div>
</div>"
`;

exports[`ObsIntegrator matches snapshot within obs 1`] = `
"<div class=\\"layout vertical center-horizontal\\">
  <!--v-if-->
</div>"
`;
